internal/server/iigointernal/judiciary.go

```diff
@@ -132,3 +151,140 @@ func softMergeSanctionThresholds(clientSanctionMap map[roles.IIGOSanctionTier]ro
 	}
 	return defaultMap
 }
+
+// getIslandSanctionTier if statement based evaluator for which sanction tier a particular score is in
+func getIslandSanctionTier(islandScore roles.IIGOSanctionScore, scoreMap map[roles.IIGOSanctionTier]roles.IIGOSanctionScore) roles.IIGOSanctionTier {
+	if islandScore <= scoreMap[roles.SanctionTier1] {
+		return roles.NoSanction
+	} else if islandScore <= scoreMap[roles.SanctionTier2] {
+		return roles.SanctionTier1
+	} else if islandScore <= scoreMap[roles.SanctionTier3] {
+		return roles.SanctionTier2
+	} else if islandScore <= scoreMap[roles.SanctionTier4] {
+		return roles.SanctionTier3
+	} else if islandScore <= scoreMap[roles.SanctionTier5] {
+		return roles.SanctionTier4
+	} else {
+		return roles.SanctionTier5
+	}
+}
+
+// getTierSanctionMap basic mapping between snaciotn tier and rule that governs it
+func getTierSanctionMap() map[roles.IIGOSanctionTier]string {
+	return map[roles.IIGOSanctionTier]string{
+		roles.SanctionTier1: "iigo_economic_sanction_1",
+		roles.SanctionTier2: "iigo_economic_sanction_2",
+		roles.SanctionTier3: "iigo_economic_sanction_3",
+		roles.SanctionTier4: "iigo_economic_sanction_4",
+		roles.SanctionTier5: "iigo_economic_sanction_5",
+	}
+}
+
+// defaultInitLocalSanctionCache generates a blank sanction cache
+func defaultInitLocalSanctionCache(depth int) map[int][]roles.Sanction {
+	returnMap := map[int][]roles.Sanction{}
+	for i := 0; i < depth; i++ {
+		returnMap[i] = []roles.Sanction{}
+	}
+	return returnMap
+}
+
+func implementPardons(sanctionCache map[int][]roles.Sanction, pardons map[int][]bool, allTeamIds []shared.ClientID) (bool, map[int][]roles.Sanction, map[shared.ClientID][]map[shared.CommunicationFieldName]shared.CommunicationContent) {
+	if validatePardons(sanctionCache, pardons) {
+		finalSanctionCache := sanctionCache
+		communicationsAboutPardons := generateEmptyCommunicationsMap(allTeamIds)
```

Comment:
is allTeamIds a global store somewhere?

Reply:
I don't think so, its fed into the function using the shared.TeamIds in the calling function.

Reply:
oh my bad completely missed. Apolgies